deepspeed/runtime/config.py

```diff
@@ -864,7 +851,13 @@ def _initialize_params(self, param_dict):
         self.gradient_predivide_factor = get_gradient_predivide_factor(param_dict)
         self.sparse_gradients_enabled = get_sparse_gradients_enabled(param_dict)
 
-        self.zero_config = DeepSpeedZeroConfig(param_dict)
+        if ZERO_OPTIMIZATION in param_dict:
```

Comment:
Refactoring this logic could make it easier to read. Something like 
`self.zero_config = get_...`